go/wire.go

```diff
@@ -217,6 +221,10 @@ func WriteFloat64(buf []byte, n float64) {
 	WriteUint64(buf, math.Float64bits(n))
 }
 
+func WriteBytes32(buf []byte, v [32]byte) {
+	*(*[32]byte)(unsafe.Pointer(&buf[0])) = v
```

Comment:
I don't understand how this works, but I'm assuming it's too C for me to grok :)

Reply:
bytes are one after the other (packed) so it is allowed to do this:
a, b := [32]bytes{}, [32]bytes{}
a = ... something
b = a (copy values to b)